custom_components/idotmatrix/__init__.py

```
@@ -284,13 +295,45 @@ async def async_set_saved_design(call):
+            if isinstance(coordinator, IDotMatrixCoordinator):
+                await coordinator.async_display_gif(
+                    path=path,
+                    rotation_interval=rotation_interval,
+                )
+
+    hass.services.async_register(DOMAIN, "display_gif", async_display_gif)
+
+    # Register stop_gif_rotation service
+    async def async_stop_gif_rotation(call):
+        """Handle the stop_gif_rotation service call."""
+        for entry_id, coordinator in hass.data[DOMAIN].items():
```

Ruff (0.15.13)

[warning] Loop control variable `entry_id` not used within loop body

Rename unused `entry_id` to `_entry_id`

(B007)